plasma/plasma-mvp/specifications/morevp.md

```diff
@@ -0,0 +1,277 @@
+
+An optimized version of the protocol is presented to account for restrictions of the Ethereum Virtual Machine. This optimization effectively caches the results of gas-heavy operations to decrease cost to the end user. A proof of the reduction from the optimized protocol to the basic protocol is provided to demonstrate that this optimized design also satisfies our formalized requirements. 
+
+We note the existence of certain attack vectors in the protocol, but find that most of these vectors can be largely mitigated and isolated to a relatively small attack surface. These attack vectors and their mitigations are described in detail. Although we conclude that the design is safe under certain reasonable assumptions about user behavior, some points are highlighted and earmarked for future consideration.
+
+Overall, the MoreVP exit protocol is a significant improvement over the original Plasma MVP exit protocol. We can further combine several optimizations to enhance user experience and reduce costs for users. Future work will focus on decreasing the implementation complexity of the design and continuing to minimize gas usage. 
+
+
+## Exit Protocol
+
+In this section, we specify the MoreVP exit protocol and give an intuitive argument toward its correctness. A formal treatment of the protocol is presented in the appendix. 
+
+
+### Definitions
+
+#### Deposit Transaction
+
+A deposit transaction is a special type of transaction that creates a new balance on the Plasma chain. This is similar to a Bitcoin "coinbase" transaction. In practice, this takes the form of a transaction with a single “special” input.
+
+
+#### Spend Transaction
+
+A spend transaction is any transaction that spends a UTXO present on the Plasma chain. Unlike a deposit transaction, a spend transaction does not create any new balance.
+
+
+#### In-flight Transaction
+
+A transaction is considered to be “in-flight” if it has been broadcast but has not yet been included in the Plasma chain. 
+
+
+#### Canonical Transaction
+
+A transaction is “canonical” if none of its inputs were previously spent in any other transaction. The definition of “previously spent” depends on whether or not the transaction in question is included in the Plasma chain.
+
+If the transaction was included in the chain, an input to that transaction would be considered previously spent if another transaction also spending the input was included in the chain *before* the transaction in question. Transaction position is determined by the tuple (block number, transaction index). If the transaction was not included in the chain, an input to that transaction would be considered previously spent if another transaction also spending the input is *known to exist*.
```

Comment:
would move "Transaction position is determined..." as the first sentence to this paragraph